tests/tst_calendarmanager/tst_calendarmanager.cpp

```diff
@@ -592,13 +566,9 @@ void tst_CalendarManager::test_notebookApi()
     QCOMPARE(mManager->defaultNotebook(), mAddedNotebooks.first()->uid());
     QCOMPARE(defaultNotebookSpy.count(), 1);
 
+    notebookSpy.clear();
     mManager->setDefaultNotebook(mAddedNotebooks.last()->uid());
-    for (int i = 0; i < 30; i++) {
-        if (notebookSpy.count() > 4)
-            break;
-
-        QTest::qWait(100);
-    }
+    QVERIFY(notebookSpy.wait());
```

Comment:
Wouldn't this be racy, in theory at least? Signal could come before wait() is started. QTRY_COMPARE() would handle that?

Reply:
Yes. I improved it but I'm not sure if it's perfect yet. At least it doesn't have that same issue now.